code/parse/sexp.cpp

```diff
@@ -17358,17 +17358,13 @@ void set_secondary_ammo (int ship_index, int requested_bank, int requested_ammo,
 	// Set the number of weapons
 	shipp->weapons.secondary_bank_ammo[requested_bank] = requested_ammo ;
 
-	// If a rearm limit has been specified set it.
-	if (rearm_limit >= 0)
+	// Don't allow more weapons than the bank can actually hold. -- Cyborg17 - no matter what.
+	if (rearm_limit > maximum_allowed) 
```

Comment:
Shouldn't this be `if (rearm_limit < 0 || rearm_limit > maximum_allowed)`?

Reply:
Absolutely.

I'm now working on matching changes to set_primary_ammo().